sqlmesh/core/linter/rules/builtin.py

```diff
@@ -4,21 +4,39 @@
 
 import typing as t
 
+from sqlglot.expressions import Star
 from sqlglot.helper import subclasses
 
-from sqlmesh.core.linter.rule import Rule, RuleViolation
+from sqlmesh.core.linter.helpers import TokenPositionDetails
+from sqlmesh.core.linter.rule import Rule, RuleViolation, Range
 from sqlmesh.core.linter.definition import RuleSet
 from sqlmesh.core.model import Model, SqlModel
 
 
 class NoSelectStar(Rule):
-    """Query should not contain SELECT * on its outer most projections, even if it can be expanded."""
+    """Query should not contain SELECT * on its outermost projections, even if it can be expanded."""
 
     def check_model(self, model: Model) -> t.Optional[RuleViolation]:
+        # Only applies to SQL models, as other model types do not have a query.
         if not isinstance(model, SqlModel):
             return None
-
-        return self.violation() if model.query.is_star else None
+        query = model.query
+        if query.is_star:
+            violation_range = self._get_range(model)
+            return self.violation(violation_range=violation_range)
+        return None
+
+    def _get_range(self, model: SqlModel) -> t.Optional[Range]:
+        """Get the range of the violation if available."""
+        try:
+            if len(model.query.expressions) == 1 and isinstance(model.query.expressions[0], Star):
```

Comment:
Should this instead call `find(Star)` on the model's query? Note that SQL like `t.*` won't be picked up by this check because the projection is a `Column` that contains a `Star` child in `this`.